modules/db/src/main/java/org/mule/module/db/internal/config/domain/database/DbConfigDefinitionParser.java

```diff
@@ -81,21 +85,45 @@ protected void doParse(Element element, ParserContext context, BeanDefinitionBui
 
     private void parseCustomDataTypes(Element element, BeanDefinitionBuilder builder)
     {
-        List<DbType> vendorDbTypes = new ArrayList<DbType>();
+        List<DbType> customDbTypes = new ArrayList<DbType>();
 
-        Element vendorTypes = DomUtils.getChildElementByTagName(element, DATA_TYPES_ELEMENT);
+        Element customTypes = DomUtils.getChildElementByTagName(element, DATA_TYPES_ELEMENT);
 
-        if (vendorTypes != null)
+        if (customTypes != null)
         {
-            for (Element dataType : DomUtils.getChildElementsByTagName(vendorTypes, DATA_TYPE_ELEMENT))
+            for (Element dataType : DomUtils.getChildElementsByTagName(customTypes, DATA_TYPE_ELEMENT))
             {
                 String name = dataType.getAttribute(TYPE_NAME_ATTIRBUTE);
                 int id = Integer.valueOf(dataType.getAttribute(TYPE_ID_ATTRIBUTE));
-                vendorDbTypes.add(new ResolvedDbType(id, name));
+                if (id == Types.ARRAY)
+                {
+                    customDbTypes.add(new ArrayResolvedDbType(id, name));
+                }
+                else if (id == Types.STRUCT)
+                {
+                    Class<?> mappedClass = null;
+                    final String className = dataType.getAttribute("className");
+                    if (!StringUtils.isEmpty(className))
+                    {
+                        try
+                        {
+                            mappedClass = Class.forName(className);
+                        }
+                        catch (ClassNotFoundException e)
+                        {
+                            throw new IllegalArgumentException("Cannot find mapped class: " + className);
+                        }
+                    }
+                    customDbTypes.add(new StructResolvedDbType(id, name,  mappedClass));
```

Comment:
Can StrunctResolvedDataType be split into 2 classes? One that has the mappedClass and ont that doesn't. The one that has it might be able to use the generic type as habdled by CallableStatement#getObject(String, Class).